src/self_driving/planning/pilot_util.cpp

```diff
@@ -214,18 +221,28 @@ std::unique_ptr<metrics::PipelineMetricRawData> PilotUtil::CollectPipelineFeatur
       //  PlanningContext to get the latest modifications (e.g. index changes). We punt on this since we focus on
       //  using the stats to get the features instead of executing queries. We might eventually delete this code path
       //  as well.
-      planning_context.GetTaskManager()->AddTask(std::make_unique<task::TaskDML>(
-          db_oid, query_text, std::make_unique<optimizer::TrivialCostModel>(),
-          std::vector<std::vector<parser::ConstantValueExpression>>(*params), std::vector<type::TypeId>(*param_types),
-          nullptr, metrics_manager, settings, true, true, std::make_optional<execution::query_id_t>(qid),
-          common::ManagedPointer(&sync)));
+      planning_context.GetTaskManager()->AddTask(
+          task::TaskDML::Builder()
+              .SetDatabaseOid(db_oid)
+              .SetQueryText(std::move(query_text))
+              .SetFuture(common::ManagedPointer(&sync))
+              // TODO(WAN): Copying of params and param_types, is that intentional?
```

Comment:
Synchronous tasks don't need copying, but asynchronous tasks may go out of the scope of the original `params` and `param_types`. If this becomes too expensive, we may be able to avoid copying for synchronous tasks by adding a new API (see the comments at the top of this code block). Addressing the life cycle issue for asynchronous tasks may be challenging though.

Reply:
I replaced the TODO with your comment, thanks!